src/compiler/transformers/transform-utils.ts

```diff
@@ -415,7 +415,7 @@ export const validateReferences = (
 ) => {
   Object.keys(references).forEach((refName) => {
     const ref = references[refName];
-    if (ref.path === '@stencil/core' && MEMBER_DECORATORS_TO_REMOVE.has(refName)) {
+    if (ref.path === '@stencil/core' && MEMBER_DECORATORS_TO_REMOVE.includes(refName as any)) {
```

Comment:
Does `refName` need to be cast as `any` here?

Reply:
yeah unfortunately, basically if you have a `ReadonlyArray` with strings like `"foo" | "bar"` as it's members then all the array methods which are generic across the type of the elements of the array (like `Array.prototype.map`, `Array.prototype.filter`, etc) inherit that type. so normally you want like `Array<T>.map` to have a type like `<F>((el: T) => F) => F[]`, but in this case it basically means that if you have a `ReadonlyArray` like

```ts
const readOnly = [ "foo", "bar" ] as const;
```

then the type of methods like `includes` will look like

```ts
(el: "foo" | "bar") => boolean
```

rather than what we'd sort of want here, which is

```ts
(el: string) => boolean
```

or maybe

```ts
(el: string) => el is "foo" | "bar"
```

so then basically you get a type error if you pass something of type `string` to the `.includes` on such an array, because `"foo" | "bar"` is a narrower type than `string`.

I put a little note in one other spot about this:

https://github.com/ionic-team/stencil/blob/aed400b1815c7328afd325bdfb5e5f63d1e07e04/src/cli/parse-flags.ts#L331-L337

I could add the same here, or maybe create a little helper to use for these things, like 

```ts
function readOnlyArrayHasStringMember<T, F extends T>(xs: ReadonlyArray<T>, member: F | string): boolean {
  return xs.includes(member as any)
}
```

Reply:
Hmm okay, I thought `includes` kept a more generic signature. Feels like it adopting a different signature defeats the purpose of the method 🤷‍♂️

Reply:
I think the utility function here that @alicewriteswrongs suggested would be a good idea, 'cause I think I asked the same question @tanner-reits did here on `parse-flags.ts`.

We could probably narrow that type assertion a smidge by asserting that `member` is `typeof xs[number]`, although TBH I'm not sure how much _actually_ helps in practice
```
function readOnlyArrayHasStringMember<T, F extends T>(xs: ReadonlyArray<T>, member: F | string): boolean {
  return xs.includes(member as typeof xs[number]);
}
```

Reply:
just pushed a change to add the little helper, lmk what y'all think